pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/KitRequestSettings.java

```diff
@@ -46,6 +46,31 @@ public class KitRequestSettings {
                     + "LEFT JOIN sub_kits_settings subK ON (subK.ddp_kit_request_settings_id = dkc.ddp_kit_request_settings_id) "
                     + "LEFT JOIN kit_type kt ON (dkc.kit_type_id = kt.kit_type_id)";
 
+    private static final String SQL_SELECT_KIT_REQUEST_SETTINGS_FOR_SUB_KIT =
+            "SELECT subK.kit_type_id, ddp_instance_id, cs_to.carrier as carrierTo, cs_to.easypost_carrier_id as carrierToId, "
+                    + " cs_to.carrier_account_number as carrierToAccountNumber, cs_to.service as serviceTo, "
+                    + "  cs_return.carrier as carrierReturn, cs_return.easypost_carrier_id as carrierReturnId, "
+                    + " cs_return.carrier_account_number as carrierReturnAccountNumber, cs_return.service as serviceReturn, "
+                    + "dim.kit_height, dim.kit_weight, dim.kit_length, dim.kit_width, dkc.collaborator_sample_type_overwrite, "
+                    + "dkc.collaborator_participant_length_overwrite, ret.return_address_name, ret.return_address_street1, "
+                    + "ret.return_address_street2, ret.return_address_city, ret.return_address_state, "
+                    + "ret.return_address_zip, ret.return_address_country, ret.return_address_phone, "
+                    + "dkc.kit_type_display_name, dkc.external_shipper, dkc.external_name, dkc.external_client_id, dkc.has_care_of, "
+                    + "subK.kit_type_id, subK.external_name, subK.kit_count, subK.hide_on_sample_pages, "
+                    + "(SELECT kit.kit_type_name FROM kit_type kit WHERE kit.kit_type_id = subK.kit_type_id) AS subKitName, "
+                    + "(SELECT count(dkc2.ddp_kit_request_settings_id) "
+                    + "FROM ddp_kit_request_settings dkc2 "
+                    + "LEFT JOIN sub_kits_settings subK ON (subK.ddp_kit_request_settings_id = dkc2.ddp_kit_request_settings_id) "
+                    + "WHERE subK.ddp_kit_request_settings_id = dkc2.ddp_kit_request_settings_id "
+                    + " AND dkc2.ddp_kit_request_settings_id = dkc.ddp_kit_request_settings_id) AS has_sub_kits "
+                    + "FROM ddp_kit_request_settings dkc "
+                    + "LEFT JOIN kit_dimension dim ON (dkc.kit_dimension_id = dim.kit_dimension_id) "
+                    + "LEFT JOIN carrier_service cs_to ON (dkc.carrier_service_to_id=cs_to.carrier_service_id) "
+                    + "LEFT JOIN carrier_service cs_return ON (dkc.carrier_service_return_id=cs_return.carrier_service_id) "
+                    + "LEFT JOIN kit_return_information ret ON (dkc.kit_return_id=ret.kit_return_id) "
+                    + "LEFT JOIN sub_kits_settings subK ON (subK.ddp_kit_request_settings_id = dkc.ddp_kit_request_settings_id) "
+                    + "LEFT JOIN kit_type kt ON (subK.kit_type_id = kt.kit_type_id) ";
+
```

Comment:
This is very close to the query `SQL_SELECT_CARRIER` , and looks very complicated but is basically joining the ddp_kit_request_settings table with all of the other shipmen tables (like return information, kit dimensions, carriers, etc ) and then the difference is that the kit type here is found by joining it with the sub_kits_settings table instead of kit_type table